lib/resolvers/project.js

```diff
@@ -103,5 +103,19 @@ module.exports = ({ models }) => ({ action, data, id, meta }, transaction) => {
       .then(() => Project.queryWithDeleted(transaction).findById(id));
   }
 
+  if (action === 'delete-amendments') {
+    return ProjectVersion.query(transaction).where({ projectId: id })
+      .then(versions => {
+        const granted = versions.find(v => v.status === 'granted');
+        return versions.filter(v => v.status === 'draft' && v.createdAt > granted.createdAt);
+      })
+      .then(recentDraftVersions => {
+        return ProjectVersion.query(transaction)
```

Comment:
you can just query the returned models directly instead of looking them up again

```js
return recentDraftVersions.$query(transaction).delete()
```

Reply:
Will `$query` still exist on the array after the `filter`?